### ISSUE TITLE
Doorstop error when reading cyrillic characters in item text (solution proposed)

### ISSUE THREAD

**Issue by 4upka** [doorstop-dev/doorstop#543]
When using doorstop v.2.1.4 from Spyder on Win10 with the following code:

```
import doorstop

tree = doorstop.build()
document = tree.find_document('DOC-REQ-SW')
for item in document:
    print(item.text)
```

on doorstop items with cyrillic characters present in item text the following error occurs:

>   File ".\appdata\local\programs\python\python37-32\lib\site-packages\doorstop\common.py", line 118, in read_text
>     raise DoorstopError(msg)
> 
> DoorstopError: reading '.\DOC-REQ-SW\DOC-REQ-SW-0010.yml' failed: 'charmap' codec can't decode byte 0x98 in position 2780: character maps to <undefined>

I managed to cure this behaviour by explicitly stating _encoding='utf-8'_ in _common.py_ file in the following way:

**_common.py_ with erroneous behaviour (from line 103 to line 118):**

```
def read_text(path):
    """Read text from a file.

    :param path: file path to read from
    :param encoding: input file encoding

    :return: string

    """
    log.trace("reading text from '{}'...".format(path))  # type: ignore
    try:
        with open(path, 'r') as f:
            return f.read()
    except Exception as e:
        msg = "reading '{}' failed: {}".format(path, e)
        raise DoorstopError(msg)
```

**corrected _common.py_ with normally expected behaviour (from line 103 to line 118):**

```
def read_text(path):
    """Read text from a file.

    :param path: file path to read from
    :param encoding: input file encoding

    :return: string

    """
    log.trace("reading text from '{}'...".format(path))  # type: ignore
    try:
        with open(path, 'r', encoding='utf-8') as f:
            return f.read()
    except Exception as e:
        msg = "reading '{}' failed: {}".format(path, e)
        raise DoorstopError(msg)
```

**Comment by 4upka** [comment]
Possibly related: [Issue 524](https://github.com/doorstop-dev/doorstop/issues/524)

**Comment by neerdoc** [comment]
I've done some tests on this and added some test files, including Cyrillic characters, to test @urbasus UTF-8 fix and it seems to be working.

However, I would just like to add that I could not replicate the errors (on linux, I did not try on Windows) even if reverting the patch (using codecs) with a correctly formatted UTF8-file. The only way I could replicate the error was if I forced a file containing UTF-8 characters to be saved in another format, and then it continued to fail even with the patch. So I'm not sure that the problem is actually related to the reading of the files, or if it is simply badly formatted file (in which case the load _should_ fail!).
